packages/transformers/src/lit-transformer.ts

```diff
@@ -0,0 +1,263 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as ts from 'typescript';
+import {LitElementMutations} from './mutations.js';
+
+type Visitor = ClassDecoratorTransformer | MemberDecoratorTransformer;
+
+interface ClassDecoratorTransformer {
+  kind: 'classDecorator';
+  decoratorName: string;
+
+  visit(
+    mutations: LitElementMutations,
+    class_: ts.ClassDeclaration,
+    decorator: ts.Decorator
+  ): void;
+}
+
+interface MemberDecoratorTransformer {
+  kind: 'memberDecorator';
+  decoratorName: string;
+
+  visit(
+    mutations: LitElementMutations,
+    member: ts.ClassElement,
+    decorator: ts.Decorator
+  ): void;
+}
+
+const unreachable = (x: never) => x;
+
+/**
+ * Configurable transformer for LitElement classes.
+ */
+export class LitTransformer {
+  private _context: ts.TransformationContext;
+
+  private _classDecoratorVisitors = new Map<
+    string,
+    ClassDecoratorTransformer[]
+  >();
+
+  private _memberDecoratorVisitors = new Map<
+    string,
+    MemberDecoratorTransformer[]
+  >();
+
+  private _removeNodes = new Set<ts.Node>();
+
+  constructor(context: ts.TransformationContext, visitors: Array<Visitor>) {
+    this._context = context;
+    for (const visitor of visitors) {
+      switch (visitor.kind) {
+        case 'classDecorator': {
+          let arr = this._classDecoratorVisitors.get(visitor.decoratorName);
+          if (arr === undefined) {
+            arr = [];
+            this._classDecoratorVisitors.set(visitor.decoratorName, arr);
+          }
+          arr.push(visitor);
+          break;
+        }
+
+        case 'memberDecorator': {
+          let arr = this._memberDecoratorVisitors.get(visitor.decoratorName);
+          if (arr === undefined) {
+            arr = [];
+            this._memberDecoratorVisitors.set(visitor.decoratorName, arr);
+          }
+          arr.push(visitor);
+          break;
+        }
+
+        default:
```

Comment:
Recommend always wrapping cases in brackets, even when there's just one statement inside

Reply:
Replaced it with `if` because there are only two cases and it's not really more readable as a `switch` imo.